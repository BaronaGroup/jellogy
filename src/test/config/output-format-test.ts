import {prepareTestKarhu} from '../test-utils'

describe('output-format-test', () => {
  describe('text', () => {
    const karhuTest = prepareTestKarhu({
      outputFormat: 'text'
    })

    it('is supported', karhuTest((karhu, output) => {
      karhu.context('test-context').info('Hello', {a: {b: 1}}, [1, 2, 3])
      expect(output.tracked).toMatchSnapshot()
    }))
  })

  describe('json', () => {
    const karhuTest = prepareTestKarhu({
      outputFormat: 'json'
    })

    it('is supported', karhuTest((karhu, output) => {
      karhu.context('test-context').info('Hello', {a: {b: {c: {d: 1}}}}, [1, 2, 3])
      expect(output.tracked).toMatchSnapshot()
    }))

    it('handles infinite recursion', karhuTest((karhu, output) => {
      const recursive: any = {},
        another = {a: recursive}
      recursive.b = another

      karhu.context('test-context').info(recursive)
      expect(output.tracked).toMatchSnapshot()
    }))

  })
})
