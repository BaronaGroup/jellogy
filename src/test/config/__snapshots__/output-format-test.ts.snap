// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output-format-test json handles infinite recursion 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"test-context\\",\\"logLevel\\":\\"INFO\\",\\"details\\":\\"[ { b: { a: [Circular] } } ]\\"}",
      "INFO",
      "test-context",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`output-format-test json is supported 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"test-context\\",\\"logLevel\\":\\"INFO\\",\\"details\\":[\\"Hello\\",{\\"a\\":{\\"b\\":{\\"c\\":{\\"d\\":1}}}},[1,2,3]]}",
      "INFO",
      "test-context",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`output-format-test text is supported 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "[1] INFO test-context Hello { a: { b: 1 } } [ 1, 2, 3 ]",
      "INFO",
      "test-context",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`output-format-test transport can override format 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"test-context\\",\\"logLevel\\":\\"INFO\\",\\"details\\":[\\"Hello\\",{\\"a\\":{\\"b\\":1}},[1,2,3]]}",
      "INFO",
      "test-context",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;
